akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRolePartitioningSpec.scala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding
+
+import akka.actor._
+import akka.cluster.sharding.ShardRegion.{ ClusterShardingStats, GetClusterShardingStats }
+import akka.cluster.MultiNodeClusterSpec
+import akka.persistence.journal.leveldb.SharedLeveldbJournal
+import akka.remote.testkit.{ MultiNodeConfig, MultiNodeSpec }
+import akka.testkit._
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.concurrent.duration._
+
+// Tests the case where cluster roles are used with cluster.min-nr-of-members, no per role min set
+// with 5 node cluster, 2 roles: 3 nodes role R1, 2 nodes role R2
+// See https://github.com/akka/akka/issues/28177#issuecomment-555013145
+object E1 {
+  val TypeKey = "Datatype1"
+  val extractEntityId: ShardRegion.ExtractEntityId = {
+    case id: String => (id, id)
+  }
+
+  val extractShardId: ShardRegion.ExtractShardId = {
+    case id: String => id
+  }
+}
+
+object E2 {
+  val TypeKey = "Datatype2"
+  val extractEntityId: ShardRegion.ExtractEntityId = {
+    case id: Int => (id.toString, id)
+  }
+
+  val extractShardId: ShardRegion.ExtractShardId = {
+    case id: Int => id.toString
+  }
+}
+
+abstract class ClusterShardingMinMembersPerRoleConfig extends MultiNodeConfig {
+  val first = role("first")
+  val second = role("second")
+  val third = role("third")
+  val fourth = role("fourth")
+  val fifth = role("fifth")
+
+  val r1Config: Config = ConfigFactory.parseString("""akka.cluster.roles = [ "R1" ]""")
+  val r2Config: Config = ConfigFactory.parseString("""akka.cluster.roles = [ "R2" ]""")
+
+  commonConfig(
+    ConfigFactory
+      .parseString("""
+        akka.cluster.min-nr-of-members = 2
```

Comment:
If we make this:
```suggestion
        akka.cluster.min-nr-of-members = 3
```
... then the test actually exercises the change in this PR

Reply:
`akka.cluster.min-nr-of-members=2` is the user case. I was going for first replicating that which I could not.